test/unit/spec/common/cellComponents/filePathWidget-Spec.js

```diff
@@ -0,0 +1,135 @@
+/*global beforeEach */
+/*global define*/ // eslint-disable-line no-redeclare
+/*global describe, expect, it*/
+/*global beforeAll, afterAll*/
+/*jslint white: true*/
+
+define([
+    'common/cellComponents/filePathWidget',
+    'jquery',
+    'common/runtime',
+    'common/props',
+    'common/spec',
+    'json!../../../../data/testAppObj.json',
+], function(
+    FilePathWidget,
+    $,
+    Runtime,
+    Props,
+    Spec,
+    TestAppObject
+) {
+    'use strict';
+
+    describe('The file path widget module', function() {
+        it('loads', function() {
+            expect(FilePathWidget).not.toBe(null);
+        });
+
+        it('has expected functions', function() {
+            expect(FilePathWidget.make).toBeDefined();
+        });
+
+    });
+
+    describe('The file path widget instance', function() {
+        beforeAll(() => {
+            Jupyter.narrative = {
+                getAuthToken: () => 'fakeToken',
+            };
+        });
+
+        afterAll(() => {
+            Jupyter.narrative = null;
+        });
+
+        let filePathWidget, node, spec, parameters;
+
+        beforeEach(function () {
+            const bus = Runtime.make().bus();
+            node = document.createElement('div');
+            document.getElementsByTagName('body')[0].appendChild(node);
+
+            const model = Props.make({
+                data: TestAppObject,
+                onUpdate: (props) => {},
+            });
+
+            spec = Spec.make({
+                appSpec: model.getItem('app.spec'),
+            });
+
+            parameters = spec.getSpec().parameters;
+
+            const workspaceId = 54745;
+
+            filePathWidget = FilePathWidget.make({
+                bus: bus,
+                workspaceId: workspaceId,
+                initialParams: model.getItem('params'),
+            });
+        });
+
+        afterEach(() => {
+            $('body').empty();
+        });
+
+        it('has a factory which can be invoked', function() {
+            expect(filePathWidget).not.toBe(null);
+        });
+
+        it('has the required methods', function() {
+            expect(filePathWidget.start).toBeDefined();
+            expect(filePathWidget.stop).toBeDefined();
+            expect(filePathWidget.bus).toBeDefined();
+        });
+
+        it('should start and render itself', () => {
+            return filePathWidget.start({
+                node: node,
+                appSpec: spec,
+                parameters: parameters,
+            }).then(() => {
+                expect(node.innerHTML).toContain('File Paths');
```

Comment:
I'd just put in an array of things that you want to look for, and then cycle through them with a `for` loop, i.e.

```js
const contents = ['File Paths', 'kb-file-path__table', '...'];
contents.forEach( (item) => {
  expect(node.innerHTML).toContain(item);
});
```

Reply:
good call